src/forms/BoundForm.tsx

```diff
@@ -0,0 +1,260 @@
+import { FieldState, ObjectState } from "@homebound/form-state";
+import { ReactNode, useMemo } from "react";
+import { LoadingSkeleton } from "src/components";
+import { Css, Only, Properties } from "src/Css";
+import { useComputed } from "src/hooks";
+import { Value } from "src/inputs/Value";
+import { TextFieldXss } from "src/interfaces";
+import { fail, safeEntries, useTestIds } from "src/utils";
+import { BoundCheckboxField, BoundCheckboxFieldProps } from "./BoundCheckboxField";
+import { BoundCheckboxGroupField, BoundCheckboxGroupFieldProps } from "./BoundCheckboxGroupField";
+import { BoundDateField, BoundDateFieldProps } from "./BoundDateField";
+import { BoundDateRangeField, BoundDateRangeFieldProps } from "./BoundDateRangeField";
+import { BoundIconCardField, BoundIconCardFieldProps } from "./BoundIconCardField";
+import { BoundIconCardGroupField, BoundIconCardGroupFieldProps } from "./BoundIconCardGroupField";
+import { BoundMultiLineSelectField, BoundMultiLineSelectFieldProps } from "./BoundMultiLineSelectField";
+import { BoundMultiSelectField, BoundMultiSelectFieldProps } from "./BoundMultiSelectField";
+import { BoundNumberField, BoundNumberFieldProps } from "./BoundNumberField";
+import { BoundRadioGroupField, BoundRadioGroupFieldProps } from "./BoundRadioGroupField";
+import { BoundRichTextField, BoundRichTextFieldProps } from "./BoundRichTextField";
+import { BoundSelectField, BoundSelectFieldProps } from "./BoundSelectField";
+import { BoundSwitchField, BoundSwitchFieldProps } from "./BoundSwitchField";
+import { BoundTextAreaField, BoundTextAreaFieldProps } from "./BoundTextAreaField";
+import { BoundTextField, BoundTextFieldProps } from "./BoundTextField";
+import { BoundToggleChipGroupField, BoundToggleChipGroupFieldProps } from "./BoundToggleChipGroupField";
+import { BoundTreeSelectField, BoundTreeSelectFieldProps } from "./BoundTreeSelectField";
+import { FormLines } from "./FormLines";
+
+type BoundFieldInputFnReturn = { component: ReactNode; minWith: Properties["minWidth"] };
+type BoundFieldInputFn<F> = (field: ObjectState<F>[keyof F]) => BoundFieldInputFnReturn;
+
+// To aid in discoverability of the optional override via IntelliSense, we can enumerate each form key `foo`
+// as `reactNodeFoo` as well as allow for any non-form key related `reactNodeBar` to be rendered as-is.
+type CapitalizeFirstLetter<S extends string> = S extends `${infer First}${infer Rest}`
+  ? `${Uppercase<First>}${Rest}`
+  : S;
+const reactNodePrefix = "reactNode";
+type TReactNodePrefix<S extends string> = `${typeof reactNodePrefix}${CapitalizeFirstLetter<S>}`;
+
+type CustomReactNodeKey = `${typeof reactNodePrefix}${string}`;
+
+type BoundFormRowInputs<F> = Partial<{
+  [K in keyof F]: BoundFieldInputFn<F>;
+}> & {
+  [K in CustomReactNodeKey]: ReactNode;
+} & {
+  [K in keyof F as TReactNodePrefix<K & string>]: ReactNode;
```

Comment:
If both `K in CustomReactNodeKey` and `K in keyof F as ...` both map to `ReactNode`s, isn't the first `K in CustomReactNodeKey` a superset of the the 2nd, i.e. we probably don't need this line 46, because anything the user would do like `reactNodeFirstName` (that "matches Author.firstName") already matched on `reactNode...whatever you want..." anyway?

Reply:
Oh, I thin you're saying in the other comment that it helps discoverability...makes sense.

Reply:
yeah exactly, got some good input from Tyler on that above - wish there was a way to tell Intellisense that `reactNode*` is a valid option from the list vs it being a missing option that happens to not type error.